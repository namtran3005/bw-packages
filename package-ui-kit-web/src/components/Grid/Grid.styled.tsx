import * as React from 'react';
import styled from 'styled-components';
import { Grid as MuiGrid } from '@material-ui/core';
import { GridProps as MuiGridProps } from '@material-ui/core/Grid';

import { RootSpacingProps } from '../Content/Content.styled';

interface GridProps {
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  fullHeight?: boolean;
  component?: React.ElementType;
  spacing?: 0 | 8 | 16 | 24 | 32 | 40 | 48 | 56 | 64;
}

type RootProps = GridProps &
  Omit<MuiGridProps, 'ref' | 'spacing'> &
  RootSpacingProps;

const RootComponent: React.FunctionComponent<RootProps> = ({
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  fullHeight,
  spacing = 0,
  ...cleanProps
}) => (
  <MuiGrid spacing={(spacing / 8) as MuiGridProps['spacing']} {...cleanProps} />
);

const Root = styled(RootComponent)`
  min-height: ${(props) => (props.fullHeight ? '100%' : 'auto')};
  box-sizing: border-box;
`;

Root.displayName = 'Grid';
RootComponent.displayName = 'GridComponent';

export { Root, RootProps, RootComponent };
