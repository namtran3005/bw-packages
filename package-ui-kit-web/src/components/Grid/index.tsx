export { Grid, GridItem, GridItemProps, GridProps } from './Grid';
