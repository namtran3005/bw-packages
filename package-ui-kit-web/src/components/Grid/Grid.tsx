import * as React from 'react';

import { GridItem, GridItemProps } from '../GridItem';
import { Root, RootProps } from './Grid.styled';

class Grid extends React.Component<RootProps> {
  public static Item = GridItem;
  public render(): JSX.Element {
    return <Root container {...this.props} />;
  }
}

export { Grid, GridItem, GridItemProps, RootProps as GridProps };
