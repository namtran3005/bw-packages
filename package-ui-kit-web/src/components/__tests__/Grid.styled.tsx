import * as React from 'react';
import { shallow } from 'enzyme';
import { Root, RootComponent } from '../Grid/Grid.styled';

describe('Grid', () => {
  it('should render Grid RootComponent', () => {
    expect(shallow(<RootComponent />)).toMatchSnapshot();
  });
  it('should render Grid', () => {
    expect(shallow(<Root fullHeight={false} />)).toMatchSnapshot();
  });
  it('should render Grid full height', () => {
    expect(shallow(<Root fullHeight />)).toMatchSnapshot();
  });
});
