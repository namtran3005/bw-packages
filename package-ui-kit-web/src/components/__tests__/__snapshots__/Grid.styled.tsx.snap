// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render Grid 1`] = `
.c0 {
  min-height: auto;
  box-sizing: border-box;
}

<GridComponent
  className="c0"
  fullHeight={false}
/>
`;

exports[`Grid should render Grid RootComponent 1`] = `
<WithStyles(ForwardRef(Grid))
  spacing={0}
/>
`;

exports[`Grid should render Grid full height 1`] = `
.c0 {
  min-height: 100%;
  box-sizing: border-box;
}

<GridComponent
  className="c0"
  fullHeight={true}
/>
`;
